src/App.tsx

```diff
@@ -2,28 +1,57 @@
+/* eslint-disable @typescript-eslint/indent */
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const TOTAL_ITEMS = 42;
+const items = getNumbers(1, TOTAL_ITEMS).map(n => `Item ${n}`);
+const itemsPerPage = [3, 5, 10, 20];
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(itemsPerPage[1]);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const itemsFrom = perPage * currentPage - perPage + 1;
```

Comment:
The calculation of the index for the first visible item is a bit complex and not immediately clear. Consider adding a comment to explain the logic or refactoring it into a more descriptive named function for better readability.